daffodil-core/src/main/scala/org/apache/daffodil/dsom/RuntimePropertyMixins.scala

```diff
@@ -124,14 +124,10 @@ trait TermRuntimeValuedPropertiesMixin
     } else
       Nope
 
-  final lazy val maybeFillByteEv = {
-    if (optionFillByteRaw.isDefined) {
-      val ev = new FillByteEv(fillByte, charsetEv, tci)
-      ev.compile(tunable)
-      One(ev)
-    } else {
-      Nope
-    }
+  final lazy val fillByteEv = {
+    val ev = new FillByteEv(fillByte, charsetEv, tci)
```

Comment:
Any thoughts on backwards compatibility? This could potentially break schemas that didn't provide ``dfdl:fillByte`` but didn't end up using fillByte. But maybe that's better than having a failure that we don't detect until runtime?

Reply:
Isn't fillByte of ParseOrUnparseState in ProcessorStateBases.scala only invoked when fill byte is needed? That's the one that would throw the SDE if fillbyte is not provided.

Reply:
Updated the tests that were failing by setting fillByte